keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlBakedImageRepository.kt

```diff
@@ -0,0 +1,54 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.keel.artifacts.BakedImage
+import com.netflix.spinnaker.keel.artifacts.DebianArtifact
+import com.netflix.spinnaker.keel.persistence.BakedImageRepository
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.BAKED_IMAGES
+import com.netflix.spinnaker.keel.sql.RetryCategory.READ
+import com.netflix.spinnaker.keel.sql.RetryCategory.WRITE
+import org.jooq.DSLContext
+import org.slf4j.LoggerFactory
+import java.time.Clock
+
+class SqlBakedImageRepository(
+  private val jooq: DSLContext,
+  private val clock: Clock,
+  private val objectMapper: ObjectMapper,
+  private val sqlRetry: SqlRetry,
+) : BakedImageRepository {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun store(image: BakedImage) {
+    log.debug("Storing baked image $image")
+    sqlRetry.withRetry(WRITE) {
+      jooq
+        .insertInto(BAKED_IMAGES)
+        .set(BAKED_IMAGES.PACKAGE_VERSION, image.appVersion)
+        .set(BAKED_IMAGES.CLOUD_PROVIDER, image.cloudProvider)
+        .set(BAKED_IMAGES.BASE_OS, image.baseOs)
+        .set(BAKED_IMAGES.BASE_LABEL, image.baseLabel.name)
+        .set(BAKED_IMAGES.TIME_DETECTED, clock.instant())
+        .set(BAKED_IMAGES.IMAGE, objectMapper.writeValueAsString(image))
+        .onDuplicateKeyIgnore()
+        .execute()
+    }
+  }
+
+  override fun getLatestByArtfiactVerstion(version: String, artifact: DebianArtifact): BakedImage? {
+     return sqlRetry.withRetry(READ) {
+      jooq
+        .select(BAKED_IMAGES.IMAGE)
+        .from(BAKED_IMAGES)
+        .where(BAKED_IMAGES.PACKAGE_VERSION.eq(version))
```

Comment:
The only condition here is matching on the version, so the `latest` in the function name was a bit misleading, I thought. Is there ever a case where we may have a dupe with a different timestamp, in which case we should sort by that and really pick the latest? Otherwise, I'd suggest dropping the `Latest` from the name.

Reply:
Oh good point. There won't be a duplicate, so I'll drop latest from the name so it's more clear.